pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -450,7 +453,7 @@ func (c *Controller) InstancesByPort(hostname model.Hostname, reqSvcPort int,
 					az, sa, uid := "", "", ""
 					if pod != nil {
 						az = c.GetPodAZ(pod)
-						sa = kubeToIstioServiceAccount(pod.Spec.ServiceAccountName, pod.GetNamespace(), c.domainSuffix)
+						sa = kubeToIstioServiceAccount(pod.Spec.ServiceAccountName, pod.GetNamespace())
```

Comment:
Same issue again - 'domainSuffix' should probably match trustDomain, it would be very confusing otherwise to configure 2 different domains. 

We may have cases where they're different - but I think by default we should use the same.

Reply:
See my comment below about ```spiffe.DetermineTrustDomain```